common/web/gesture-recognizer/src/engine/headless/gestures/matchers/gestureMatcher.ts

```diff
@@ -0,0 +1,293 @@
+
+    this.predecessor = predecessor;
+    this.publishedPromise = new ManagedPromise();
+
+    this.model = model;
+    if(model.sustainTimer) {
+      this.sustainTimerPromise = new TimeoutPromise(model.sustainTimer.duration);
+      this.sustainTimerPromise.then((elapsed) => {
+        const shouldResolve = model.sustainTimer.expectedResult == elapsed;
+        this.finalize(shouldResolve, 'timer');
+      });
+    }
+
+    this.pathMatchers = [];
+
+    let sourceTouchpoints: SimpleGestureSource<Type>[];
+    if(source) {
+      sourceTouchpoints = source.touchpoints;
+    } else {
+      sourceTouchpoints = predecessor.pathMatchers.map((matcher) => matcher.source);
+    }
+
+    let offset = 0;
+    for(let touchpointIndex = 0; touchpointIndex < sourceTouchpoints.length; touchpointIndex++) {
+      const srcContact = sourceTouchpoints[touchpointIndex];
+
+      // If a touchpoint's path is already complete, ignore it when modeling a new gesture.
+      if(srcContact.path.isComplete) {
+        offset++;
+        continue;
+      }
+
+      if(srcContact instanceof SimpleGestureSourceSubview) {
+        srcContact.disconnect();  // prevent further updates from mangling tracked path info.
+      }
+      let i = touchpointIndex - offset;
+
+      const contactSpec = model.contacts[i];
+      /* c8 ignore next 3 */
+      if(!contactSpec) {
+        throw new Error(`No contact model for inherited path: gesture "${model.id}', entry ${i}`);
+      }
+      const inheritancePattern = contactSpec?.model.pathInheritance ?? 'chop';
+
+      let preserveBaseItem: boolean = false;
+
+      let contact: SimpleGestureSource<Type>;
+      switch(inheritancePattern) {
+        case 'reject':
+          this.finalize(false, 'path');
+          return;
+        case 'full':
+          contact = srcContact.constructSubview(false, true);
+          this.addContactInternal(contact);
+          continue;
+        case 'partial':
+          preserveBaseItem = true;
+          // Intentional fall-through
+        case 'chop':
+          contact = srcContact.constructSubview(true, preserveBaseItem);
+          this.addContactInternal(contact);
+          break;
+      }
+    }
+  }
+
+  private finalize(matched: boolean, cause: FulfillmentCause) {
+    if(this.publishedPromise.isFulfilled) {
+      return this._result;
+    }
+
+    try {
+      // Determine the correct action-spec that should result from the finalization.
+      let action: GestureResolutionSpec | (RejectionDefault & ResolutionItemSpec);
+      if(matched) {
+        // Easy peasy - resolutions only need & have the one defined action type.
+        action = this.model.resolutionAction;
+      } else {
+        // Some gesture types may wish to restart with a new base item if they fail due to
+        // it changing during its lifetime or due to characteristics of the contact-point's
+        // path.
+        if(this.model.rejectionActions && this.model.rejectionActions[cause]) {
```

Comment:
```suggestion
        if(this.model.rejectionActions?.[cause]) {
```

it's pretty (or is it?) but definitely a nit

Reply:
🙏 

I've been _wanting_ this!  Thanks!  Didn't realize there actually was a pattern that let it work.